book/src/user/shielded-scan-grpc-server.md

```diff
@@ -0,0 +1,148 @@
+# Zebra Shielded Scanning gRPC Server
+
+## Get Started
+
+### Setup
+
+After setting up [Zebra Shielded Scanning](https://zebra.zfnd.org/user/shielded-scan.html), add a `listen_addr` field to the shielded-scan configuration:
+
+```toml
+[shielded_scan]
+listen_addr = "127.0.0.1:8231"
+```
+
+Then, run `zebrad` to start the scan gRPC server.
+
+Making requests to the server will also require a gRPC client, the examples here use `grpcurl`, though any gRPC client should work.
+
+[See installation instructions for `grpcurl` here](https://github.com/fullstorydev/grpcurl?tab=readme-ov-file#installation).
+
+The types can be accessed through the `zebra-grpc` crate's root `scanner` module for clients in a Rust environment, and the [`scanner.proto` file here](https://github.com/ZcashFoundation/zebra/blob/main/zebra-grpc/proto/scanner.proto) can be used to build types in other environments.
+
+### Usage
+
+To check that the gRPC server is running, try calling `scanner.Scanner/GetInfo`, for example with `grpcurl`:
+
+```bash
+grpcurl -plaintext '127.0.0.1:8231' scanner.Scanner/GetInfo
+```
+
+The response should look like:
+
+```
+{
+  "minSaplingBirthdayHeight": 419200
+}
+```
+
+An example request to the `Scan` method with `grpcurl` would look like: 
+
+```bash
+grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/Scan
+```
+
+This will start scanning for transactions in Zebra's state and in new blocks as they're validated.
+
+Or, to use the scanner gRPC server without streaming, try calling `RegisterKeys` with your Sapling extended full viewing key, waiting for the scanner to cache some results, then calling `GetResults`:
+
+```bash
+grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/RegisterKeys
```

Comment:
It seems this just works even if the sapling key is invalid?

```
alfredo@spaceship:~/zebra$ ./grpcurl -plaintext -d '{ "keys": { "key": ["sapling_extended_full_viewing_key"] } }' '127.0.0.1:8231' scanner.Scanner/RegisterKeys
{}
alfredo@spaceship:~/zebra$ 
```

We get the same responseif the key is valid.
```
alfredo@spaceship:~/zebra$ ./grpcurl -plaintext -d '{ "keys": { "key": ["zxviews1q0duytgcqqqqpqre26wkl45gvwwwd706xw608hucmvfalr759ejwf7qshjf5r9aa7323zulvz6plhttp5mltqcgs9t039cx2d09mgq05ts63n8u35hyv6h9nc9ctqqtue2u7cer2mqegunuulq2luhq3ywjcz35yyljewa4mgkgjzyfwh6fr6jd0dzd44ghk0nxdv2hnv4j5nxfwv24rwdmgllhe0p8568sgqt9ckt02v2kxf5ahtql6s0ltjpkckw8gtymxtxuu9gcr0swvz"] } }' '127.0.0.1:8231' scanner.Scanner/RegisterKeys
{}
alfredo@spaceship:~/zebra$ 
```

Reply:
It returns a list of keys that were registered, maybe it should return an error if no keys were added?

Reply:
Yea ... lets work on it in the context of https://github.com/ZcashFoundation/zebra/issues/8316